src/loaders/muse_load.c

```diff
@@ -0,0 +1,126 @@
+/* Extended Module Player
+ * Copyright (C) 1996-2022 Claudio Matsuoka and Hipolito Carraro Jr
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "loader.h"
+#include "../miniz.h"
+#include "../md5.h"
+
+static void set_md5sum(unsigned char *src, unsigned char *md5, size_t sz)
+{
+	MD5_CTX ctx;
+	MD5Init(&ctx);
+	MD5Update(&ctx, src, sz);
+	MD5Final(md5, &ctx);
+}
+
+static int muse_test(HIO_HANDLE *, char *, const int);
+static int muse_load(struct module_data *, HIO_HANDLE *, const int);
+
+const struct format_loader libxmp_loader_muse = {
+	"MUSE container",
+	muse_test,
+	muse_load
+};
+
+static int muse_test(HIO_HANDLE * f, char *t, const int start)
+{
+	uint8 in[8];
+	uint32 r;
+
+	if (hio_read(in, 1, 8, f) != 8) {
+		return -1;
+	}
+	if (memcmp(in, "MUSE", 4) != 0) {
+		return -1;
+	}
+	r = readmem32b(in + 4);
+	/* MOD2J2B uses 0xdeadbabe */
```

Comment:
there's two versions of mod2j2b: One for the new format and one for the old format. 0xDEADBEAF indicates that the content is an RIFF AM file, 0xDEADBABE indicates that it's a RIFF AMFF file.

Reply:
gal4_load.c and gal5_load.c handle them correctly, no?

Reply:
What I mean is that this comment doesn't make any sense. modj2b writes *both* magic byte variants, it just depends on which version you use. I'd either remove the comment entirely or update it to be correct.

Reply:
Oh, I see - just removed the comment. Thanks!